### PR TITLE
Fixing for windows

### PR DESCRIPTION
**Pull request by scphantm**
Expands on Vampire's code and fixes more bugs associated with running on windows

### CODE REVIEW

**Comment by scphantm** [comment]
As i said before, you may want to consider adding appveyor to the checks routine for all of Nebula.  Unfortunately, I'm one of the poor lost souls that have to run on Winblows and any time i crack Nebula open, I encounter a Winblows related error.  I think I got them all out of this, but the gradle tests stuff is horribly old, don't want to get into that and very few unit tests check for gradle 3.x compatibility.  That stuff is beyond me at the moment tho.  Its a shame because I have some major overhauls that I could do to this library to significantly enhance its capabilities, Its just got to be brought up to the 3.x stuff, my stuff would be hard pressed to work on 2.x

**Comment by scphantm** [comment]
And if we could please get this in as quickly as possible, i would appreciate it.  I had to make these changes to address bugs I found in Nebula-Publish.  I can't get those changes merged until these changes are merged and released and the Nebula-Publish issues have me dead in the water at the moment.  Please let me know what i can do to speed this up.  Thanks

**Comment by scphantm** [comment]
It looks like when you merge this, you will also have to update nebula-plugin-plugin to pull the correct version.  In my testing of nebula-publish i have to do some pretty nasty things to get around plugin-plugins dependencies.

**Comment by scphantm** [comment]
Its been 6 days, can someone give me a status on this MR?  Code review?  Something?  This has me at a stand still and I'm running out of busy work.  Thanks

**Comment by scphantm** [comment]
in case your wondering, i need this merged and nebula-plugin updated so i can finish this PR  https://github.com/nebula-plugins/nebula-publishing-plugin/pull/71

**Comment by szpak** [comment]
@scphantm In this project PRs quite often wait long time before (eventually) merged. Waiting 6 months for my #68 to be merged I've switched my project to a [JitPack](http://jitpack.io/) version. Just as a workaround better than snapshots...

**Comment by scphantm** [comment]
thanks @szpak , im hoping i can avoid that, as everything is so interconnected, i would basically have to pull nearly all of nebula in house and start running my own builds.  I REALLY don't want to do that.

**Comment by rspieldenner** [comment]
Our new plugins are planned to use gradle testKit for testing so as we migrate more and more of them over to that, this project will become more and more helpers for testKit.

**Comment by scphantm** [comment]
That's great.  But can we get this issue clears in the mean time so I can finish fixing publish?

File: .gitignore [hunk 1]
```diff
@@ -8,3 +8,4 @@ gradle.properties
 *.iml
 *.iws
 out/
+test/
```
Comment (rspieldenner) [inline .gitignore]: Remove this, I'll fix the test that is writing outside of the build directory

File: src/test/groovy/nebula/test/SpecifiedGradleVersionIntegrationSpec.groovy [hunk 2]
```diff
@@ -23,7 +24,12 @@ class SpecifiedGradleVersionIntegrationSpec extends IntegrationSpec {
         when:
             def result = runTasksSuccessfully('build')
         then:
-            result.standardOutput.contains("gradle/$requestedGradleVersion/taskArtifacts")
+            if (Os.isFamily(Os.FAMILY_WINDOWS)){
```
Comment (rspieldenner) [inline src/test/groovy/nebula/test/SpecifiedGradleVersionIntegrationSpec.groovy]: There is a spock construct to do this. I'd rather have 2 separate tests that require different OS's then an if block in the test.

http://spockframework.org/spock/docs/1.0/extensions.html

IgnoreIf block

```
os Information about the operating system (see spock.util.environment.OperatingSystem)
```

File: .gitignore (hunk 1 above)
Comment (scphantm) [inline .gitignore]: k

File: src/test/groovy/nebula/test/SpecifiedGradleVersionIntegrationSpec.groovy (hunk 2 above)
Comment (scphantm) [inline src/test/groovy/nebula/test/SpecifiedGradleVersionIntegrationSpec.groovy]: oh i know the ignoreif annotation, i just hate the dup code is all.  but if you want two tests, no problem, i can do that.  And as i write this, i just realized how to pull it off.  give me a few minutes

Comment (scphantm) [inline src/test/groovy/nebula/test/SpecifiedGradleVersionIntegrationSpec.groovy]: ok, so my brilliant idea wasn't so brilliant.  its 2 tests now.

**Pull request merged**